pkg/intermediate/aggregate.go

```diff
@@ -482,13 +482,27 @@ func (a *AggregationProcess) aggregateRecords(incomingRecord, existingRecord ent
 		return nil
 	}
 	isLatest := false
+	var prevFlowEndSecondsFromNode, flowEndSecondsDiff uint32
 	if ieWithValue, _, exist := incomingRecord.GetInfoElementWithValue("flowEndSeconds"); exist {
 		if existingIeWithValue, _, exist2 := existingRecord.GetInfoElementWithValue("flowEndSeconds"); exist2 {
-			incomingVal := ieWithValue.GetUnsigned32Value()
-			existingVal := existingIeWithValue.GetUnsigned32Value()
-			if incomingVal > existingVal {
+			newFlowEndSeconds := ieWithValue.GetUnsigned32Value()
+			prevFlowEndSeconds := existingIeWithValue.GetUnsigned32Value()
+			if newFlowEndSeconds >= prevFlowEndSeconds {
 				isLatest = true
-				existingIeWithValue.SetUnsigned32Value(incomingVal)
+				existingIeWithValue.SetUnsigned32Value(newFlowEndSeconds)
+			}
+			// Update the flowEndSecondsFromSource/DestinationNode fields, and compute
+			// the time difference between the incoming record and the last record.
+			if fillSrcStats {
+				prevFlowEndSecondsFromNode = a.updateFlowEndSecondsFromNodes(incomingRecord, existingRecord, "flowEndSecondsFromSourceNode", newFlowEndSeconds)
+			}
+			if fillDstStats {
+				prevFlowEndSecondsFromNode = a.updateFlowEndSecondsFromNodes(incomingRecord, existingRecord, "flowEndSecondsFromDestinationNode", newFlowEndSeconds)
+			}
+			flowEndSecondsDiff = newFlowEndSeconds - prevFlowEndSecondsFromNode
```

Comment:
Since `flowEndSecondsDiff` is uint32 type, it would not be smaller than 0. How about changing these lines to:
```
if newFlowEndSeconds < prevFlowEndSecondsFromNode {
   return nil
}
flowEndSecondsDiff = newFlowEndSeconds - prevFlowEndSecondsFromNode
```

Reply:
updated. That's right, thanks for catching it.